platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/CameraChangeDispatcher.java

```diff
@@ -15,10 +14,10 @@
 
   private boolean idle = true;
 
-  private final List<OnCameraMoveStartedListener> onCameraMoveStartedListenerList = new ArrayList<>();
-  private final List<OnCameraMoveCanceledListener> onCameraMoveCanceledListenerList = new ArrayList<>();
-  private final List<OnCameraMoveListener> onCameraMoveListenerList = new ArrayList<>();
-  private final List<OnCameraIdleListener> onCameraIdleListenerList = new ArrayList<>();
+  private final CopyOnWriteArrayList<OnCameraMoveStartedListener> onCameraMoveStarted = new CopyOnWriteArrayList<>();
```

Comment:
@tobrun Adding `CopyOnWriteArrayList` makes it thread-safe, but may result in a rare scenario when user unregisters the listener, yet still gets the callback. Do you think we should add synchronization?

Reply:
I haven't ran into any issues with that or haven't heard about any reports. The reason why I don't perceive this as an issue is that the interactions with this list allows occurs on the main thread. 
The reason why we have a CopyOnWriteArrayList is to allow to remove the listener as part of it's callback to avoid a `ConcurrentModificationException`. Imo your remark is a tradeoff of this, happy to look more into this if there is a use-case for it.

Reply:
The problem arises when you'd like to register/unregister on a separate thread (which in my scenario would have thrown  `ConcurrentModificationException` before and now will just execute the callback). Since your use-case is concrete and the problem was there before anyway, just like in a different form, I think we're good here.

>The reason why I don't perceive this as an issue is that the interactions with this list allows occurs on the main thread.

Maybe adding `@UiThread` annotation when registering in `MapboxMap` would be nice if we expect users to register/unregister on the main thread?

Reply:
We annotate the class with `@UiThread`, which is the equivalent to adding the annotation to each method of the class. Would love to see if we can improve the implementation in future iterations ;)

Reply:
>We annotate the class with `@UiThread`

Missed that, totally makes sense :)